packages/ai-jsx/src/batteries/sidekick/platform/conversation.tsx

```diff
@@ -84,15 +84,7 @@ export async function getNextConversationStep(
   switch (lastMessage.type) {
     case 'functionCall': {
       const { name, args } = lastMessage.element.props;
-      try {
-        return <FunctionResponse name={name}>{await tools[name].func(args)}</FunctionResponse>;
-      } catch (e: any) {
-        return (
-          <FunctionResponse failed name={name}>
-            {e.message}
-          </FunctionResponse>
-        );
-      }
+      return <ExecuteFunction func={tools[name].func} name={name} args={args} />;
```

Comment:
I like how you managed to get around the issue of passing `getContext` down to `func`. I'm not sure if I can do the same though. I'll take another stab at it.